sync/syncer.go

```diff
@@ -17,7 +17,7 @@ import (
 )
 
```

Comment:
this was intended for internal package use i think we should keep it that way
and if possible from sync point of view the api should be 
func BlockValid(b block) bool
we can discuss this in person

Reply:
@almogdepaz just so you don't think I ignored this comment - Anton and I discussed it and it will be addressed in a separate PR on Sunday.

The interface you want (and I agree it's the right one) requires putting the eligibility proofs in the block and implementing a bit more logic, so we wanted to separate it from this PR.